1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,346 @@
+- HistoryAndPlugin - This option uses both for predictions
+
+To enable Predictive IntelliSense, enter the following command in the shell or in the users Profile:
+
+```powershell
+Set-PSReadLineOption -PredictionSource History
+```
+
+To disable Predictive IntelliSense, enter the following command in the shell or in the users Profile:
+
+```powershell
+Set-PSReadLineOption -PredictionSource None
+```
+
+### Changing the Prediction View
+
+Beginning with 2.2.0-beta1, Predictions are displayed in one of two views depending on the user
+preference. The default view is InlineView.
+
+- InlineView – This is the default view and displays the prediction inline with the user’s typing.
+  This view is similar to other shells Fish and ZSH.
+- ListView – ListView provides a dropdown list of predictions below the line the user is typing.
+  Users may quickly scan the list, highlight and select the desired prediction.
+
+Users may change the view at the command line using the keybinding F2 or `Set-PSReadLineOption`. The
+`Set-PSReadLineOption` may be stored in the user's profile.
+
+```powershell
+Set-PSReadLineOption -PredictionViewStyle ListView
+```
+
+### Change the Prediction Color for Accessibility
+
+By default, predictions appear on the same line the user is typing in a different customizable
+color. To support accessibility needs, the prediction color is settable in the shell or user's
+profile.
+
+The default light-grey prediction text color;
+
+```powershell
+Set-PSReadLineOption -Colors @{ InLinePrediction = "$([char]0x1b)[38;5;238m"}
+```
+
+Starting with PSReadLine 2.1.0-RC1, `Set-PSReadLineOption -Colors` array will include
+**InLinePrediction** to set the color of predictive text for **InLineView**.
+
+Beginning with PSReadLine 2.2.0-beta1, `Set-PSReadLineOption -Colors` array will include;
+
+- **ListPredictionColor** to set color for '>' and source names in **ListView**
+- **ListPredictionSelectionColor** to set color for the highlighted selection in **ListView**
+
+Examples of user changing prediction color:
+
+![pi-color](./media/pi-color.png)
+![pi-color](./media/pi-color2.png)
+![pi-color](./media/pi-color3-background.png)
+
+```powershell
+Set-PSReadLineOption -Colors @{ InLinePrediction = '#8A0303'}
+Set-PSReadLineOption -Colors @{ InLinePrediction = '#2F7004'}
+Set-PSReadLineOption -Colors @{ InLinePrediction = "$([char]0x1b)[36;7;238m"}
+```
+
+Multiple types of color code values are supported in PSReadLine. For more information see
+Set-PSReadLineOption in
+[PSReadLine](https://docs.microsoft.com/en-us/powershell/module/PSReadline/Set-PSReadlineOption?view=powershell-7)
+
+Examples of different color code values:
+
+```powershell
+ # Use a ConsoleColor enum
+Set-PSReadLineOption -Colors @{ InLinePrediction = 'DarkRed'}
+Set-PSReadLineOption -Colors @{ InLinePrediction = [ConsoleColor]::DarkRed}
+
+ # 24 bit color escape sequence
+Set-PSReadLineOption -Colors @{ InLinePrediction = "$([char]0x1b)[38;5;100m"}
```

Comment:
Only on those terminals that support 24 a bit, right?

Reply:
yes. I changed to # 24 bit color escape sequence on supported terminals